### ISSUE TITLE
[Suggestion] Reward command

### ISSUE THREAD

**Issue by Ghmmy** [garbagemule/MobArena#643]
# Feature request

**Short description**
Currently we arent able to make custom mobs drop special items to the players as it will clear their inventory upon finishing the arena. One way to allow those custom mobs to drop items is to add a command to mobarena which gives an item to a player after finishing the arena, just like the wave/boss-rewards, except it gets activated by command instead of completing a wave or killing a boss. Don't worry about implementing per-mob rewards as that could easily be done through MythicMob if the command-reward gets added (by simply adding an onDeath trigger that performs /ma reward <player> <reward> to that mob)

**Example:**
Steve kills a mythicmob, also known as "Frozen Zombie". the frozen zombie has a mythicmobs skill that performs a command on death: '/ma reward Steve cmd(1 Frozen Sword):/mi sword frozen_sword'. Upon killing that mob, it doesnt give the item to Steve instantly, except it tells steve the same way the wave/boss-rewards do: it puts a msg in chat telling steve he got a reward. After leaving the arena, steve gets the Frozen Sword, just like the wave/boss-rewards.

**Implementation details**

It could be done by adding a new command to mobarena (ma reward <player> <reward>; example: /ma Ghmmy cmd(1 Token):/tm add Ghmmy token 1 ), that contains a requirement of that player to be in an arena, otherwise the command cannot be performed since the player isnt in an arena. 

This would probably just require the same code thats being used for the boss- and wave-rewards, except it gets triggered by a command rather than an action. It would also need a requirement to it, as players must be in an arena for the command to work.

**Additional info**

tl;dr
Add a command "/ma reward <player> <reward>" that rewards the player after finishing the arena. It's just the command and you dont have to worry about writing a per-mob reward we could set up in the config, as MythicMob already handles that.

**Comment by garbagemule** [comment]
I quite like the generality of this idea, and indeed how non-intrusive the implementation would be. It would allow for hooking into the rewards system from a lot of different areas as long as that area allows for invoking commands and knows which player to target.

I'm pretty sure this is possible to implement without changing _any_ existing code, it's just a new command. It would be an admin command, i.e. only ops and people with the permission would have access to it by default. I think this name/format is appropriate considering the rest of the commands in the plugin:

    /ma addreward <player> <thing>

where `<thing>` just follows the normal syntax from the [Item and Rewards Syntax](../wiki/Item-and-Reward-Syntax) page, i.e. all arguments after the `<player>` argument at joined by a space and then thrown into the ThingParser.

Gonna hit it with the old "good first issue" label and leave it up for grabs if anyone wants to take a stab at it - there are plenty of commands to draw inspiration from, and there are plenty of places where we parse Things. A ThingParser instance can be grabbed via the ArenaMaster instance, and the arena's RewardManager should be available from the Arena instance, which can be fetched by player from the ArenaMaster instance as well.
